backbone.js

```diff
@@ -223,28 +167,129 @@
 
   // Implement fancy features of the Events API such as multiple event
   // names `"change blur"` and jQuery-style event maps `{change: action}`
-  // in terms of the existing API.
-  var eventsApi = function(obj, action, name, rest) {
-    if (!name) return true;
-
-    // Handle event maps.
-    if (typeof name === 'object') {
-      for (var key in name) {
-        obj[action].apply(obj, [key, name[key]].concat(rest));
+  // in terms of the existing API. Consider this just a special purpose
+  // reduce function.
+  var eventsApi = function(api, events, name, callback, context, ctx) {
+    var i = 0, names, length;
+    if (name && typeof name === 'object') {
+      // Handle event maps.
+      for (names = _.keys(name), length = names.length; i < length; i++) {
+        events = api(events, names[i], name[names[i]], context, ctx);
       }
-      return false;
+    } else if (name && eventSplitter.test(name)) {
+      // Handle space separated event names.
+      for (names = name.split(eventSplitter), length = names.length; i < length; i++) {
+        events = api(events, names[i], callback, context, ctx);
+      }
+    } else {
+      events = api(events, name, callback, context, ctx);
+    }
+    return events;
+  };
+
+  // Handles actually adding the event handler.
+  var onApi = function(events, name, callback, context, ctx) {
+    if (callback) {
+      var handlers = events[name] || [];
+      events[name] = handlers.concat({callback: callback, context: context, ctx: context || ctx});
     }
+    return events;
+  };
+
+  // Handles removing any event handlers that are no longer wanted.
+  var offApi = function(events, name, callback, context) {
+    // Remove all callbacks for all events.
+    if (!events || !name && !context && !callback) return;
+
+    var names = name ? [name] : _.keys(events);
+    for (var i = 0, length = names.length; i < length; i++) {
+      name = names[i];
 
-    // Handle space separated event names.
-    if (eventSplitter.test(name)) {
-      var names = name.split(eventSplitter);
-      for (var i = 0, length = names.length; i < length; i++) {
-        obj[action].apply(obj, [names[i]].concat(rest));
+      // Bail out if there are no events stored.
+      var handlers = events[name];
+      if (!handlers) break;
```

Comment:
It's the same kind of logic as the other. The only way the conditional will ever be true is if you pass the `name` argument, in which case `names = [name]`. So, no need to continue.